dotcom-rendering/src/web/components/Picture.tsx

```diff
@@ -1,323 +1,332 @@
+					{ breakpoint: breakpoints.phablet, width: 620 },
+				];
+		}
 	}
-
-	return desiredWidths;
 };
 
 /**
- * Returns the desired width for an image at the specified breakpoint, based on the image role, format, and if it's main media.
- * This function acts as a source of truth for widths of images based on the provided parameters
+ * Generates a url for calling the Fastly Image Optimiser
  *
- * Returns 'breakpoint', or 'breakpoint / x' when the image is expected to fill a % of the viewport,
- * instead of a fixed width.
+ * @see {@link https://developer.fastly.com/reference/io/}
  *
- * @param breakpoint
- * @param role
- * @param isMainMedia
- * @param format
- * @returns {Pixel}
  */
-const getDesiredWidthForBreakpoint = (
-	breakpoint: number,
-	role: RoleType,
-	isMainMedia: boolean,
-	format: ArticleFormat,
-): Pixel => {
-	if (
-		(format.display === ArticleDisplay.Showcase ||
-			format.display === ArticleDisplay.NumberedList) &&
-		isMainMedia
-	) {
-		// Showcase main media images (which includes numbered list articles) appear
-		// larger than in body showcase images so we use a different set of image sizes
-		if (breakpoint >= breakpoints.wide) return 1020;
-		if (breakpoint >= breakpoints.leftCol) return 940;
-		if (breakpoint >= breakpoints.tablet) return 700;
-		if (breakpoint >= breakpoints.phablet) return 700;
-		return breakpoint;
-	}
+const generateSignedUrl = ({
+	master,
+	imageWidth,
+	resolution,
+}: {
+	master: string;
+	imageWidth: number;
+	resolution: 'low' | 'high';
+}): string => {
+	const sign = (salt: string, path: string): string => {
+		return createHash('md5')
+			.update(salt + path)
+			.digest('hex');
+	};
 
-	if (isMainMedia && format.display === ArticleDisplay.Immersive) {
-		// Immersive main media images are often reported to have poor quality so we
-		// override the default values here and force larger images to be used
-		switch (breakpoint) {
-			case breakpoints.mobile:
-				return 480;
-			case breakpoints.mobileLandscape:
-				return 660;
-			case breakpoints.phablet:
-				return 740;
-			case breakpoints.tablet:
-				return 980;
-			case breakpoints.desktop:
-				return 1140;
-			case breakpoints.leftCol:
-				return 1300;
-			case breakpoints.wide:
-				return 1900;
-		}
+	const salt = process.env.IMAGE_SALT;
+	if (!salt && process.env.NODE_ENV === 'production' && !process.env.CI) {
+		// If no IMAGE_SALT env variable is set in production then something has
+		// gone wrong and we want to know about it
+		throw new Error(
+			'No IMAGE_SALT value found when constructing picture source',
+		);
+	} else if (!salt) {
+		// If no IMAGE_SALT is set outside of production it likely means the
+		// developer has yet to set the env file. In this case we continue
+		// using a default image
+		return 'https://i.guim.co.uk/img/media/52f8e9183bbc03c48fcdf4ed9a4e12b3a19a9927/0_0_5000_3000/master/5000.jpg?width=700&quality=45&auto=format&fit=max&dpr=2&s=196c7f25bf2fddcb45512d3ff25e20c0';
```

Comment:
This isn't something for this PR but in the future I'd love to have a bunch of signed urls for all possible image widths and DPRs so this reflects the 'correct' resolutions wherever it's rendered!